coil-base/src/main/java/coil/disk/DiskCache.kt

```diff
@@ -115,8 +125,18 @@ interface DiskCache {
          * IMPORTANT: It is an error to have two [DiskCache] instances active in the same
          * directory at the same time as this can corrupt the disk cache.
          */
-        fun directory(directory: File) = apply {
+        fun directory(directory: File) =
+            directory(directory = directory.toOkioPath(), fileSystem = FileSystem.SYSTEM)
+
+        /**
+         * Set the [directory] where the cache stores its data.
+         *
+         * IMPORTANT: It is an error to have two [DiskCache] instances active in the same
+         * directory at the same time as this can corrupt the disk cache.
+         */
+        fun directory(directory: Path, fileSystem: FileSystem = FileSystem.SYSTEM) = apply {
```

Comment:
I'd vote to have a separate `DiskCache.Builder.fileSystem` function instead of a default argument.